### PR TITLE
feat: support elastic index prefix

### PR DESCRIPTION
**Pull request by eruizgar91**
## Description

Add a description of your changes here.

## Is this PR related with an open issue?

Related to Issue #

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] Follows the code style of this project.
- [ ] Tests Cover Changes
- [ ] Documentation

### CODE REVIEW

File: .env.sample [hunk 1]
```diff
@@ -1,6 +1,6 @@
 JWT_SECRET_KEY="secret"
 NODE_ENV="development"
 ELASTIC_URL="http://localhost:9200"
-ELASTIC_INDEX="marketplace"
+ELASTIC_PREFIX="marketplace"
```
Comment (aaitor) [inline .env.sample:4]: probably the name should be a bit more explicit. Like:
`ELASTIC_INDEX_NAME_PREFIX`

File: config/development.js [hunk 2]
```diff
@@ -14,6 +14,7 @@ module.exports = {
   },
   elasticsearch: {
     node: process.env.ELASTIC_NODE,
+    prefix: process.env.ELASTIC_PREFIX,
```
Comment (aaitor) [inline config/development.js:17]: change name

File: config/local.js.sample [hunk 3]
```diff
@@ -14,6 +14,7 @@ module.exports = {
   },
   elasticsearch: {
     node: 'http://localhost:9200',
+    prefix: 'marketplace',
```
Comment (aaitor) [inline config/local.js.sample:17]: default should be empty

File: src/shared/config/config.service.ts [hunk 4]
```diff
@@ -27,6 +27,7 @@ const DOTENV_SCHEMA = Joi.object({
   }),
   elasticsearch: Joi.object({
     node: Joi.string().default('http://localhost:9200'),
+    prefix: Joi.string(),
```
Comment (aaitor) [inline src/shared/config/config.service.ts:30]: empty by default

File: src/shared/elasticsearch/elastic.service.ts [hunk 5]
```diff
@@ -28,7 +32,7 @@ export class ElasticService {
     /* eslint-disable-next-line @typescript-eslint/no-unsafe-return, @typescript-eslint/no-unsafe-member-access */
     return (
       await this.elasticsearchService.search({
-        index,
+        index: this.configService.get<string>('elasticsearch.prefix') + index,
```
Comment (dancespiele) [inline src/shared/elasticsearch/elastic.service.ts:35]: here the same

File: src/shared/elasticsearch/elastic.service.ts [hunk 6]
```diff
@@ -53,22 +57,22 @@ export class ElasticService {
   async getDocumentByIndexAndId(index: string, id: string): Promise<unknown> {
     return (
       await this.elasticsearchService.get({
-        index,
+        index: this.configService.get<string>('elasticsearch.prefix') + index,
```
Comment (dancespiele) [inline src/shared/elasticsearch/elastic.service.ts:60]: here the same

File: src/shared/elasticsearch/elastic.service.ts [hunk 7]
```diff
@@ -53,22 +57,22 @@ export class ElasticService {
...
         id,
       })
     ).body
   }
 
   async deleteDocumentByIndexAndId(index: string, id: string): Promise<unknown> {
     return this.elasticsearchService.delete({
-      index,
+      index: this.configService.get<string>('elasticsearch.prefix') + index,
```
Comment (dancespiele) [inline src/shared/elasticsearch/elastic.service.ts:68]: here the same

File: src/shared/elasticsearch/elastic.service.ts [hunk 8]
```diff
@@ -77,14 +81,15 @@ export class ElasticService {
...
       body,
     })
   }
 
   async checkExistingIndex(index: string): Promise<ApiResponse<boolean, unknown>> {
+    console.log(this.configService.get<string>('elasticsearch.prefix'))
     return this.elasticsearchService.indices.exists({
-      index,
+      index: this.configService.get<string>('elasticsearch.prefix') + index,
```
Comment (dancespiele) [inline src/shared/elasticsearch/elastic.service.ts:92]: here the same

File: src/shared/elasticsearch/elastic.service.ts [hunk 9]
```diff
@@ -77,14 +81,15 @@ export class ElasticService {
 
   async createIndex(index: string, body: unknown): Promise<void> {
     await this.elasticsearchService.indices.create({
-      index,
+      index: this.configService.get<string>('elasticsearch.prefix') + index,
```
Comment (dancespiele) [inline src/shared/elasticsearch/elastic.service.ts:84]: here the same

**Review by dancespiele — CHANGES_REQUESTED** [review]

**Review by dancespiele — APPROVED** [review]

**Pull request merged**